.github/workflows/ocr-action.yml

```
@@ -0,0 +1,75 @@
+jobs:
+  ocr-and-analyze:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Setup system deps
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y poppler-utils tesseract-ocr libtesseract-dev imagemagick python3-venv build-essential 
+      - name: Setup Python
+        uses: actions/setup-python@v4
```

actionlint (1.7.9)

the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)